proxy/src/telemetry/event.rs

```diff
@@ -100,18 +100,19 @@ impl Event {
         }
     }
 
-    pub fn proxy(&self) -> &Arc<ctx::Proxy> {
+    /// Return the proxy context associated with the event, or `None` if the
+    /// event was not associated with either proxy.
+    pub fn proxy(&self) -> Option<&Arc<ctx::Proxy>> {
```

Comment:
Since this function doesn't currently appear to be used, perhaps it should just be removed instead? I wasn't sure whether it was there because we expected to use it any time soon or not...